DatadogRUM/Sources/RUM+Internal.swift

```diff
@@ -9,6 +9,42 @@ import DatadogInternal
 
 extension RUM: InternalExtended {}
 
+extension RUM.Configuration {
+    /// Used to perform late initialization of URLSessionTracking. This is needed in order to allow override of
+    /// properties normally provided by the base `RUM.Configuration` object
+    public struct LateURLSessionTracking {
+        /// See RUM.Configuration.URLSessionTracking.firstPartyHostsTracing
+        public var firstPartyHostsTracing: RUM.Configuration.URLSessionTracking.FirstPartyHostsTracing?
+
+        /// See RUM.Configuration.URLSessionTracking.resourceAttributesProvider
+        public var resourceAttributesProvider: RUM.ResourceAttributesProvider?
+
+        // MARK: - Internal
+
+        internal var debugSDK: Bool = ProcessInfo.processInfo.arguments.contains(LaunchArguments.Debug)
+        internal var dateProvider: DateProvider = SystemDateProvider()
+        internal var traceIDGenerator: TraceIDGenerator = DefaultTraceIDGenerator()
+
+        /// Private init to avoid `invalid redeclaration of synthesized memberwise init(...:)` in extension.
+        private init() {}
+
+        public init(
+            firstPartyHostsTracing: RUM.Configuration.URLSessionTracking.FirstPartyHostsTracing? = nil,
+            resourceAttributesProvider: RUM.ResourceAttributesProvider? = nil
+        ) {
+            self.firstPartyHostsTracing = firstPartyHostsTracing
+            self.resourceAttributesProvider = resourceAttributesProvider
+        }
+
+        public init(from: URLSessionTracking) {
+            self.firstPartyHostsTracing = from.firstPartyHostsTracing
+            self.resourceAttributesProvider = from.resourceAttributesProvider
+        }
+    }
```

Comment:
/blocker The internal properties should be injected by the main config, alongside the `URLSessionTracking`. You also don't need the public accessor when using the main config. It can be something like:
```swift
// MARK: - Internal

internal let debugSDK: Bool
internal let dateProvider: DateProvider
internal let traceIDGenerator: TraceIDGenerator

public init(
    firstPartyHostsTracing: RUM.Configuration.URLSessionTracking.FirstPartyHostsTracing? = nil,
    resourceAttributesProvider: RUM.ResourceAttributesProvider? = nil,
    debugSDK: Bool = false // if you ever need to debug from CP
) {
    self.firstPartyHostsTracing = firstPartyHostsTracing
    self.resourceAttributesProvider = resourceAttributesProvider
    self.debugSDK = debugSDK
    self.dateProvider = SystemDateProvider()
    self.traceIDGenerator = DefaultTraceIDGenerator()
}

internal init(
    from configuration: URLSessionTracking,
    debugSDK: Bool,
    dateProvider: DateProvider,
    traceIDGenerator: TraceIDGenerator
) {
    self.firstPartyHostsTracing = configuration.firstPartyHostsTracing
    self.resourceAttributesProvider = configuration.resourceAttributesProvider
    self.debugSDK = debugSDK
    self.dateProvider = dateProvider
    self.traceIDGenerator = traceIDGenerator
}
```